daemon/cmd/state.go

```diff
@@ -92,11 +92,18 @@ func (d *Daemon) validateEndpoint(ep *endpoint.Endpoint) (valid bool, err error)
 				return
 			}
 
-			ep.UpdateVisibilityPolicy(func(ns, podName string) (proxyVisibility string, err error) {
-				_, _, annotations, err := d.fetchK8sLabelsAndAnnotations(ns, podName)
-				if err != nil {
-					return "", err
+			cp, _, _, annotations, err := d.fetchK8sLabelsAndAnnotations(ep.K8sNamespace, ep.K8sPodName)
```

Comment:
This seems to introduce a regression as well since the annotations are not the ones up to dates when the `UpdateVisibilityPolicy` is executed.

Reply:
Regression reverted.